lib/sb3_schema.json

```diff
@@ -87,98 +99,152 @@
+                {"type": "string", "description": "broadcast message id"}
+            ],
+            "additionalItems": false
+        },
+        "variable_primitive": {
+            "type": "array",
+            "items": [
+                {
+                    "type": "number",
+                    "enum": [12]
+                },
+                {"type": "string", "description": "variable name"},
+                {"type": "string", "description": "variable id"}
+            ],
+            "additionalItems": {
+                "type": "number"
+            },
+            "minItems": 3,
+            "maxItems": 5
+        },
+        "list_primitive": {
+            "type": "array",
+            "items": [
+                {
+                    "type": "number",
+                    "enum": [13]
+                },
+                {"type": "string", "description": "list name"},
+                {"type": "string", "description": "list id"}
+            ],
+            "additionalItems": {
+                "type": "number"
+            },
+            "minItems": 3,
+            "maxItems": 5
+        },
+        "topLevelPrimitive": {
+            "oneOf": [
+                {"$ref":"#/definitions/variable_primitive"},
+                {"$ref":"#/definitions/list_primitive"}
+            ]
+        },
+        "inputPrimitive": {
+            "oneOf": [
+                {"$ref":"#/definitions/num_primitive"},
+                {"$ref":"#/definitions/color_primitive"},
+                {"$ref":"#/definitions/text_primitive"},
+                {"$ref":"#/definitions/broadcast_primitive"},
+                {"$ref":"#/definitions/variable_primitive"},
+                {"$ref":"#/definitions/list_primitive"}
+            ]
         },
         "block": {
             "type": "object",
             "properties": {
-                "id": {
-                    "type": "string"
-                },
                 "opcode": {
                     "type": "string"
                 },
                 "inputs": {
                     "type": "object",
                     "additionalProperties": {
-                        "type": "object",
-                        "properties": {
-                            "name": {
-                                "type": "string"
-                            },
-                            "block": {"$ref":"#/definitions/optionalString"},
-                            "shadow": {"$ref":"#/definitions/optionalString"},
-                            "value": {
-                                "oneOf": [
-                                    {"type": "string"},
-                                    {"type": "number"}
-                                ]
+                        "type": "array",
+                        "items": [
+                            {
+                                "type":"number",
+                                "enum":[1,2,3]
```

Comment:
~Are these type numbers documented somewhere?~
I see that they're documented in LLK/scratch-vm#1031 but it would be nice to have them documented in this repository as well, somehow.